front/lib/upsert_document.ts

```diff
@@ -37,10 +37,23 @@ export const EnqueueUpsertDocument = t.type({
   upsertContext: t.union([UpsertContextSchema, t.null]),
 });
 
+export const EnqueueUpsertTableFromCsv = t.type({
```

Comment:
Let's not use io-ts type for function arguments, instead we can create proper TS types like this:

```
type EnqueueUpsertTableFromCsvType = t.TypeOf<typeof EnqueueUpsertTableFromCsv>;
```

I checked the call sites and the validation already happened there.

Reply:
We need this io-ts type across boundaries api / lib / temporal (e.g. there is a decode in the activity)

It's not used for validation, we do use `t.TypeOf<typeof EnqueueUpsertTableFromCsv>` already in this pr

Is your suggestion to move the io-ts elsewehere? e.g. in types?

Reply:
My point is that io-ts should not use to type function parameters but only network calls/ files retrieval.

Reply:
But that's what done here? let's talk irl